R/oauth-cache.R

```diff
@@ -0,0 +1,245 @@
+## this file has its origins in oauth-cache.R in httr
+## nothing there is exported, so copied over, then it evolved
+
+# cache --------------------------------------------------------------
+
+gargle_default_oauth_cache_path <- "~/.R/gargle/gargle-oauth"
+
+## this is the cache setup interface for the Gargle2.0 class
+cache_establish <- function(cache = getOption("gargle.oauth_cache")) {
+  if (length(cache) != 1) {
+    stop("Cache should be length 1 vector", call. = FALSE)
+  }
+  if (!is.logical(cache) && !is.character(cache)) {
+    stop("Cache must either be logical or string (file path)")
+  }
+
+  # If NA, propose default cache file
+  # request permission to create it, if doesn't exist yet
```

Comment:
Do we need to handle the case where the user can't write to that file?

Reply:
Well, it is "handled" I think because we will soon fail to create the cache in an obvious way. User can then try again, specifying the path to a place where they **can** write and if that's a long-term preference, they can store that in the option `gargle.oauth_cache`.

Do I understand your question?

Reply:
Yes — I think I was just confused by "request permission", but you mean from the user.